engine/table/src/main/java/io/deephaven/engine/table/impl/by/OperatorAggregationStateManagerTypedBase.java

```diff
@@ -55,119 +47,98 @@
+    protected final ArrayBackedColumnSource<?>[] mainKeySources;
+    // the location of the first overflow entry in this bucket, parallel to keySources
+    protected final IntegerArraySource mainOverflowLocationSource = new IntegerArraySource();
 
     // the keys for overflow
     private int nextOverflowLocation = 0;
+
+    // the overflow chains, logically a linked list using integer pointers into these three parallel array sources
     protected final ArrayBackedColumnSource<?>[] overflowKeySources;
-    // the location of the next key in an overflow bucket
+    // the location of the next key in an overflow bucket, parallel with overflowKeySources
     protected final IntegerArraySource overflowOverflowLocationSource = new IntegerArraySource();
-    // the overflow buckets for the state source
-    @HashTableAnnotations.OverflowStateColumnSource
-    // @StateColumnSourceType@ from \QIntegerArraySource\E
-    protected final IntegerArraySource overflowStateSource
-    // @StateColumnSourceConstructor@ from \QIntegerArraySource()\E
-            = new IntegerArraySource();
+
 
     protected OperatorAggregationStateManagerTypedBase(ColumnSource<?>[] tableKeySources, int tableSize,
             double maximumLoadFactor, double targetLoadFactor) {
-        // region super
-        // endregion super
-
         this.tableSize = tableSize;
         Require.leq(tableSize, "tableSize", MAX_TABLE_SIZE);
         Require.gtZero(tableSize, "tableSize");
         Require.eq(Integer.bitCount(tableSize), "Integer.bitCount(tableSize)", 1);
         this.tableHashPivot = tableSize;
 
+        mainKeySources = new ArrayBackedColumnSource[tableKeySources.length];
         overflowKeySources = new ArrayBackedColumnSource[tableKeySources.length];
-        keySources = new ArrayBackedColumnSource[tableKeySources.length];
 
         for (int ii = 0; ii < tableKeySources.length; ++ii) {
             // the sources that we will use to store our hash table
-            keySources[ii] = ArrayBackedColumnSource.getMemoryColumnSource(tableSize, tableKeySources[ii].getType());
-
+            mainKeySources[ii] =
+                    ArrayBackedColumnSource.getMemoryColumnSource(tableSize, tableKeySources[ii].getType());
             overflowKeySources[ii] =
-                    ArrayBackedColumnSource.getMemoryColumnSource(CHUNK_SIZE, tableKeySources[ii].getType());
+                    ArrayBackedColumnSource.getMemoryColumnSource(0, tableKeySources[ii].getType());
         }
 
         this.maximumLoadFactor = maximumLoadFactor;
         this.targetLoadFactor = targetLoadFactor;
-
-        ensureCapacity(tableSize);
     }
 
-    private void ensureCapacity(int tableSize) {
-        stateSource.ensureCapacity(tableSize);
-        overflowLocationSource.ensureCapacity(tableSize);
-        for (int ii = 0; ii < keySources.length; ++ii) {
-            keySources[ii].ensureCapacity(tableSize);
+    protected void ensureCapacity(int tableSize) {
```

Comment:
`mainEnsureCapacity`? Consider following the same pattern as overflow, for symmetry, or annotating with `javax.annotation.OverridingMethodsMustInvokeSuper`.